brigand/algorithms/detail/replace.hpp

```diff
@@ -0,0 +1,46 @@
+/*==================================================================================================
+  Copyright (c) 2015 Edouard Alligand and Joel Falcou
+  Author: Marek Kurdej
+
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+=================================================================================================**/
+#pragma once
+
+#include <brigand/functions/lambda/apply.hpp>
+#include <brigand/types/bool.hpp>
+#include <type_traits>
+
+#ifdef _MSC_VER
+#define BRIGAND_DEDUCED_TEMPLATE template
```

Comment:
Yeah, MSVC2013, with 2015 it's fine... Completely illogical for me.

Error from MSVC2013:
```
3>C:\workspace\brigand\brigand/algorithms/detail/replace.hpp(35): error C2825: 'brigand::apply': must be a class or namespace when followed by '::'
3>          C:\workspace\brigand\brigand/algorithms/detail/replace.hpp(38) : see reference to class template instantiation 'brigand::detail::predicate_test<Predicate,T,NewType>' being compiled
```